bin-src/tasks/gitInfo.ts

```diff
@@ -175,6 +176,17 @@ export const setGitInfo = async (ctx: Context, task: Task) => {
             .join('\n')}`
         );
       }
+
+      const changedPackageManifests = ctx.git.changedFiles.filter((fileName) =>
+        [/^package\.json$/, /\/package\.json$/].some((re) => re.test(fileName))
+      );
+
+      if (changedPackageManifests.length > 0) {
+        ctx.git.packageManifestsWithDependencyChanges = await getPackageManagerChanges(
+          ctx.build,
+          changedPackageManifests
+        );
+      }
```

Comment:
@ghengeveld should I be looking over all the baseline commits here (instead of just the most recent), similar to our approach today?

Reply:
Good point. What we should be doing is check against the baseline build where the manifest file was actually changed. We [currently take the union](https://github.com/chromaui/chromatic-cli/blob/83f281d2cdcc05df6fa5db97bd664b5ba819a65b/bin-src/tasks/gitInfo.ts#L157-L163) of changed files from each baseline as the total set of `changedFiles`. If `package.json` is in there, we don't know whether it was changed on one baseline, the other, or both.

I suggest we introduce a new `ctx.git.changedPackageManifestsByCommit` value which contains a Map of commit -> changed manifest files. We'd create that in the same [place](https://github.com/chromaui/chromatic-cli/blob/83f281d2cdcc05df6fa5db97bd664b5ba819a65b/bin-src/tasks/gitInfo.ts#L157-L163) where we set `ctx.git.changedFiles` right now.

Reply:
I went ahead and accounted for multiple baselines. I'm only storing the file names that changed (not the commits), since we only need to know that there are some package files changed (at the `gitInfo` level I'm caring about commits of course, so we can checkout the files to diff them).